src/neptune/api/exceptions_utils.py

```diff
@@ -35,8 +35,8 @@ def handle_json_errors(
     if error_processor:
         raise error_processor(content) from source_exception
 
-    if default_exception and isinstance(source_exception, BaseException):
+    if default_exception and source_exception is not None:
```

Comment:
We still may want to raise a `default_exception`. Just without a cause (`from`).